packages/data/src/plugins/persistence/index.js

```diff
@@ -223,75 +223,60 @@ function persistencePlugin( registry, pluginOptions ) {
 }
 
 /**
- * Deprecated: Remove this function and the code in WordPress Core that calls
- * it once WordPress 5.4 is released.
+ * Move the 'features' object in local storage from the sourceStoreName to the
+ * interface store.
+ *
+ * @param {*} persistence
+ * @param {*} sourceStoreName
  */
-
-persistencePlugin.__unstableMigrate = ( pluginOptions ) => {
-	const persistence = createPersistenceInterface( pluginOptions );
-
+export function migrateFeaturePreferencesToInterfaceStore(
+	persistence,
+	sourceStoreName
+) {
+	const interfaceStoreName = 'core/interface';
 	const state = persistence.get();
+	const sourcePreferences = state[ sourceStoreName ]?.preferences;
+	const sourceFeatures = sourcePreferences?.features;
+
+	if ( sourceFeatures ) {
+		const targetFeatures =
+			state[ interfaceStoreName ]?.preferences?.features;
+
+		if ( ! targetFeatures ) {
+			// Set the feature values in the interface store, the features
+			// object is keyed by 'scope', which matches the store name for
+			// the source.
+			persistence.set( interfaceStoreName, {
+				preferences: {
+					features: {
+						[ sourceStoreName ]: sourceFeatures,
```

Comment:
Will we at a later stage be consolidating the features that are to be shared across editors (not necessarily all of them) under a "shared" key or something?

Reply:
This aspect wasn't something I'd considered when working on this task, but it's absolutely possible after this change now that the values are stored in one place. 

I think the challenge will be how we reconcile that a user might already have conflicting values for preferences across editors. If I have Top Toolbar active in the post editor, but not active in customize widgets, which one is right?

I don't think the implementation of a shared preference would be difficult now, I can think of a few different ways to do it, it all depends on how we'd want it to work. Would it be as simple as one value to rule them all, or might we allow a combination of a global and local preference?

Reply:
These are all great points. 

Regarding global vs local, I was thinking for some options - mostly the a11y/usability related ones, such as text labels on buttons and keeping the cursor inside the block - it make sense to just set them globally, as it's unlikely someone would want to use them in only one editor. For the others, maybe a global option with local override, or local-only makes more sense. 

I initially thought of site settings as the best place to put global options, but might be good to get some design input on that too.